functions/Test-DbaDatabseBestPractice

```diff
@@ -0,0 +1,305 @@
+					$dbfiles += New-Object PSObject -Property @{
+						Server = $server.name
+						Database = $db.Name
+						FileGroupName = $filegroup.name
+						FileGroupType = $filegroup.FileGroupType
+						FileSize = $file.Size
+						GrowthType = $file.GrowthType
+						GrowthSize = $file.Growth
+						MaxSize = $maxsize
+						FileType = "Data"
+						IsBestPractice = $isbestpractice
+					}
+				}
+			}
+			
+			foreach ($file in $db.LogFiles)
+			{
+				$bpgrowthsize = $file.Size * 0.125
+				$isbestpractice = ($file.GrowthType -eq "Percent" -and $file.Growth -ge $bpgrowthsize)
+				
+				if ($file.MaxSize -eq -1)
+				{
+					$maxsize = "Unlimited"
+				}
+				else
+				{
+					$maxsize = $file.MaxSize
+					$maxsize = "$maxsize KB"
+				}
+				
+				$dbfiles = New-Object PSObject -Property @{
+					Server = $server.name
+					Database = $db.Name
+					FileGroupName = $file.name
+					FileGroupType = "Not Applicable" 
+					FileSize = $file.Size
+					GrowthType = $file.GrowthType
+					GrowthSize = $file.Growth
+					MaxSize = $maxsize
+					FileType = "Log"
+					IsBestPractice = $isbestpractice
+				}
+			}
+			
+			return $dbfiles
+		}
+
+		$collection = New-Object System.Collections.ArrayList
+	}
+	
+	PROCESS
+	{
+        $processed = New-Object System.Collections.ArrayList
+
+		foreach ($server in $sqlserver) 
+		{
+            if ($server -notin $processed)
+			{
+				$null = $processed.Add($server)
+				Write-Verbose "Connecting to $server"
+			}
+			else
+			{
+				continue
+			}
+
+			Write-Verbose "Attempting to connect to $sqlserver"
+			
+			try
+			{
+				$server = Connect-SqlServer -SqlServer $sqlserver -SqlCredential $SqlCredential
+			}
+			catch
+			{
+				Write-Warning "Can't connect to $sqlserver or access denied. Skipping."
+				continue
+			}
+			
+            $databases = $server.Databases;
+
+			if ($databases.count -eq 0)
+			{
+				$dbs = $server.Databases | Where-Object { $databases -contains $_.Name }
+			}
+			else
+			{
+				if ($IncludeSystemDbs -eq $true)
+				{
+					$dbs = $server.Databases
+				}
+				else
+				{
+					$dbs = $server.Databases | Where-Object { $systemdbs -notcontains $_.Name }
+				}
+			}
+			
+			foreach ($db in $dbs)
+			{
+				$autogrowthbps = Test-AutoGrowth $server, $db
+				$falsecount = $autogrowthbps | Where-Object { $_.IsBestPractice -eq $false }
+				
+				if ($falsecount.count -gt 0)
+				{
+					$IsBestPracticeAutoGrowth = $false
+				}
+				else
+				{
+					$IsBestPracticeAutoGrowth = $true
+				}
+				
+				$object = New-Object PSObject -Property @{
+					Server = $sqlserver
+					Database = $db.Name
+					IsBestPracticeAutoClose = Test-AutoClose $server $db
+					IsBestPracticeAutoShrink = Test-AutoShrink $server, $db
+					IsBestPracticePageVerify = Test-PageVerify $server, $db
+					IsBestPracticeAutoGrowth = $IsBestPracticeAutoGrowth
+					AutoGrowthDetails = $autogrowthbps
+				}
+				$null = $collection.Add($object)
+			}
+		}
+	}
+	
+	END
+	{
+		if ($Detailed -eq $true)
```

Comment:
The detailed output results in file autogrowth details causing other stuff to be chopped off the screen, only visible if you output to grid etc. Consider setting the order of the output to Server, Database, IsBestPractice[n] and putting filegrowth detail at the end. This is similar to the order mentioned in the description.